source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -28,17 +28,40 @@ FileImplPosix::~FileImplPosix() {
   }
 }
 
-void FileImplPosix::openFile() {
-  const int flags = O_RDWR | O_APPEND | O_CREAT;
-  const int mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
-
-  fd_ = ::open(path_.c_str(), flags, mode);
+void FileImplPosix::openFile(FlagSet in) {
+  std::tuple<int, int> flags_set = translateFlag(in);
```

Comment:
Instead of returning a tuple, can you return a struct or something easier to read? Same below.

Reply:
Return a struct instead of tuple.